### PR TITLE
[POC] Separate Demo integrations 

### PR DESCRIPTION
**Pull request by descorp**
### Changes

* separate Demo code into "DropIn" and "Components" integrations
* improve parity between UIKit and SwiftUI
* refactor naming and folder structure

### CODE REVIEW

**Review by mohammedDehairy — DISMISSED** [review]

File: Demo/Common/Helpers/HasAPIClient.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+//  HasAPIClient.swift
+//  Adyen
+//
+//  Created by Vladimir Abramichev on 05/02/2021.
+//  Copyright © 2021 Adyen. All rights reserved.
+//
+
+import Adyen
+
+protocol HasAPIClient {
```
Comment (mohammedDehairy) [inline Demo/Common/Helpers/HasAPIClient.swift:9]: `APIClientAware` makes more sense to me, its sort of a convention.

File: Demo/Common/Integrations/ComponentIntegrationController.swift [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        currentComponent?.didCancel()
+
+        if let paymentComponent = self.currentComponent as? PaymentComponent {
+            paymentComponent.delegate?.didFail(with: ComponentError.cancelled, from: paymentComponent)
+        }
+    }
+
+    // MARK: - Standalone Components
+
+    internal func Card() {
```
Comment (mohammedDehairy) [inline Demo/Common/Integrations/ComponentIntegrationController.swift:69]: starts with capital `C`.

File: Demo/Common/Integrations/ComponentIntegrationController.swift [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                                      configuration: CardComponent.Configuration(),
+                                      clientKey: Configuration.clientKey,
+                                      style: formStyle)
+        component.environment = environment
+        component.clientKey = Configuration.clientKey
+        component.cardComponentDelegate = self
+        present(component)
+    }
+
+    internal func Ideal() {
```
Comment (mohammedDehairy) [inline Demo/Common/Integrations/ComponentIntegrationController.swift:81]: starts with capital.

File: Demo/Common/Integrations/ComponentIntegrationController.swift [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        guard let paymentMethod = paymentMethods?.paymentMethod(ofType: IssuerListPaymentMethod.self) else { return }
+        let component = IdealComponent(paymentMethod: paymentMethod,
+                                       style: listStyle)
+        present(component)
+    }
+
+    internal func SEPADirectDebit() {
```
Comment (mohammedDehairy) [inline Demo/Common/Integrations/ComponentIntegrationController.swift:88]: starts with capital.

File: Demo/Common/Integrations/ComponentIntegrationController.swift [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+    internal func MBWay() {
+        guard let paymentMethod = paymentMethods?.paymentMethod(ofType: MBWayPaymentMethod.self) else { return }
+        let component = MBWayComponent(paymentMethod: paymentMethod,
+                                       style: formStyle)
+        component.delegate = self
+        present(component)
+    }
+
+    internal func ApplePay() {
```
Comment (mohammedDehairy) [inline Demo/Common/Integrations/ComponentIntegrationController.swift:104]: starts with capital.

File: Demo/Common/Integrations/ComponentIntegrationController.swift [hunk 6]
```diff
@@ -0,0 +1,174 @@
+//
+//  ComponentIntegrationController.swift
+//  Adyen
+//
+//  Created by Vladimir Abramichev on 05/02/2021.
+//  Copyright © 2021 Adyen. All rights reserved.
+//
```
Comment (mohammedDehairy) [inline Demo/Common/Integrations/ComponentIntegrationController.swift:5]: I thought we don't use this kind of header anymore.

File: Demo/Common/Integrations/DropInIntegrationController.swift [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import AdyenComponents
+import UIKit
+
+internal final class DropInIntegrationController: AdyenIntegrationController {
+
+    // MARK: - DropIn Component
+
+    private lazy var dropInComponentStyle = DropInComponent.Style()
+
+    internal func DropIn() {
```
Comment (mohammedDehairy) [inline Demo/Common/Integrations/DropInIntegrationController.swift:20]: starts with capital.

**Comment by mohammedDehairy** [comment]
i think it will be confusing to merchants, they would have to jump between 3-4 files to figure out the whole integration steps.

before it was simpler:

1. A Controller contains all the logic.
2. diffferent presenters for UIKit and SwIftUI.

now its `AdyenIntegrationController`, `DropInIntegrationController`, `ComponentIntegrationController`,  and `ComponentsModel `.

if you want my opinion, i think its too much for an example code :-D 
if this was production code, it would have been good, but not for example code.

File: Demo/Common/Integrations/ComponentIntegrationController.swift (hunk 6 above)
Comment (descorp) [reply Demo/Common/Integrations/ComponentIntegrationController.swift:5]: nope, cleaned

**Pull request closed without merging**